common/txmgr/abandoned_tracker.go

```diff
@@ -0,0 +1,194 @@
+package txmgr
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/exp/slices"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	txmgrtypes "github.com/smartcontractkit/chainlink/v2/common/txmgr/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+// TTL is the default time to live for abandoned transactions (6hrs)
+const TTL = 6 * time.Hour
+
+// AbandonedErrorMsg occurs when an abandoned tx exceeds its time to live
+var AbandonedErrorMsg = fmt.Sprintf(
+	"abandoned transaction exceeded time to live of %d hours", int(TTL.Hours()))
+
+// AbandonedTx is a transaction who's fromAddress was removed from the Confirmer's enabledAddresses list
+type AbandonedTx[
+	CHAIN_ID types.ID,
+	ADDR types.Hashable,
+	TX_HASH types.Hashable,
+	BLOCK_HASH types.Hashable,
+	SEQ types.Sequence,
+	FEE feetypes.Fee,
+] struct {
+	tx *txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]
+	// fatalTime represents the time at which this transaction is to be marked fatal
+	fatalTime time.Time
+}
+
+// AbandonedTracker tracks and handles abandoned transactions
+type AbandonedTracker[
+	CHAIN_ID types.ID,
+	ADDR types.Hashable,
+	TX_HASH types.Hashable,
+	BLOCK_HASH types.Hashable,
+	R txmgrtypes.ChainReceipt[TX_HASH, BLOCK_HASH],
+	SEQ types.Sequence,
+	FEE feetypes.Fee,
+] struct {
+	txStore *txmgrtypes.TxStore[ADDR, CHAIN_ID, TX_HASH, BLOCK_HASH, R, SEQ, FEE]
+	ks      *txmgrtypes.KeyStore[ADDR, CHAIN_ID, SEQ]
```

Comment:
I don't think you'll need the key store in this component. The only think we'd need it for is getting the list of enabled keys in `evm.key_states` but since this is a sub-component of the Confirmer, we can just pass that list straight in.